plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/callables/FunctionSignature.kt

```diff
@@ -5,12 +5,17 @@
 
 package org.jetbrains.kotlin.formver.embeddings.callables
 
+import org.jetbrains.kotlin.fir.symbols.FirBasedSymbol
 import org.jetbrains.kotlin.formver.embeddings.TypeEmbedding
+import org.jetbrains.kotlin.formver.embeddings.expression.FirVariableEmbedding
 import org.jetbrains.kotlin.formver.embeddings.expression.VariableEmbedding
 
 interface FunctionSignature : CallableSignature {
     val receiver: VariableEmbedding?
-    val params: List<VariableEmbedding>
+    val params: List<FirVariableEmbedding>
+    val symbolsToParams: Map<FirBasedSymbol<*>?, FirVariableEmbedding>
+        get() = params.associateBy { it.symbol }
```

Comment:
If I understand correctly, this will recompute this map every time it's used.  I think it would be better to have a function that does this (e.g. an extension method of `FunctionSignature`).